onnxscript/function_libs/torch_aten/ops/core.py

```diff
@@ -3572,17 +3572,175 @@ def aten_max_pool1d_with_indices(
     raise NotImplementedError()
 
 
+@torch_op("aten::max_pool2d", trace_only=True)
 def aten_max_pool2d(
-    self: TensorType,
+    self: Union[DOUBLE, FLOAT, FLOAT16, INT8, UINT8],
     kernel_size: Sequence[int],
-    stride: Optional[Sequence[int]] = None,
+    stride: Sequence[int] = None,
     padding: Sequence[int] = (0, 0),
     dilation: Sequence[int] = (1, 1),
     ceil_mode: bool = False,
-) -> TensorType:
+) -> Union[DOUBLE, FLOAT, FLOAT16, INT8, UINT8]:
     """max_pool2d(Tensor self, int[2] kernel_size, int[2] stride=[], int[2] padding=0, int[2] dilation=1, bool ceil_mode=False) -> Tensor"""
 
-    raise NotImplementedError()
+    self_len = len(self.shape)
+    expand_size = 4
```

Comment:
I  think we don't need a default value for expand so that it can handle an input whose length is not in [3, 4].
This function is specifically designed for max_pool2d which requires an input whose length is 3 or 4.  For any other case, I think we should fail the conversion as soon as possible instead of having a default value to make the conversion pass but ort fail later.

Reply:
set expand_size = 2